docs/tutorials/developer-journey/level-2/2.3-advanced-canister-calls.md

```diff
@@ -0,0 +1,252 @@
+
+### Example update call
+
+Update calls are not defined with a function modifier in Motoko code such as query calls are. Below is a simple update call that counts the number of characters within the inputted string, updates the canister's state, then returns a `true` or `false` value if that number of characters is divisible by 2. 
+
+```motoko
+actor countCharacters {
+    public func test(text : Text) : async Bool {
+        let size = Text.size(text);
+        return size % 2 == 0;
+    };
+};
+```
+
+## Certified variables
+
+Since query calls do not go through consensus, their response cannot be validated as being accurate or secure. To remedy this, certified variables can be used. Certified variables enable queries to return an authenticated response that can be verified and trusted. 
+
+Certified variables utilize **chain-key cryptography** to generate digital signatures that can then be validated using a single, permanent public key that belongs to the IC. This is a bit different from traditional signatures, since the private key never exists in a single location; it is constantly distributed between many different nodes. Valid signatures can only be generated when the majority of the nodes storing the pieces of the private key cooperate in a cryptographic protocol. Through this method, an application can immediately validate all certified variables that it receives from a query call without having to put trust into the particular node that it received the response from. 
+
+Certification happens at the canister level. Certified variables can be set during an update call to a canister. Recall that an update call changes the canister's state and goes through consensus. When a certified variable is set, the certification can then be read in future query calls and return the certified variable in response to the query call in a trustworthy and secure manner. 
+
+If you want to dive a bit deeper, you can take a look at an the DFINITY example project using certified variables [here](https://github.com/dfinity/examples/tree/master/motoko/cert-var), or view a community example of an application using certified variables [here](https://github.com/nomeata/motoko-certified-http/blob/master/main.mo).
+
+## Composite queries
+
+A composite query call is a type of query that can only be invoked via an ingress message, such as one generated by an agent such as a web browser, or through `dfx`. Composite queries can be used for scenarios such as frontend code that wants to get information from the backend code. A great example of this is a partitioned key-value store setup, where there is one frontend canister and several different backend canisters, with each backend representing a partition of the key-value stores. Composite queries can be very useful by providing efficient data retrieval for dapps that scale vertically with data partitioned across multiple canisters.
```

Comment:
The use case of composite queries is not clear form this description. Composite queries allow a canister to call a query of another canister inside an ingress query. In your example, it would allow to have storage canisters shared by keys, and an index canister with a composite query that fetches the information from the proper shard given a key. Before composite queries, the client would have to get the information about which storage canister to contact from the index canister and do the query call directly.